staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1beta1/types_test.go

```diff
@@ -0,0 +1,97 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+)
+
+// TestCluster ensures that Cluster stays in sync with clientcmd.Cluster.
+//
+// We want Cluster to offer the same knobs as clientcmdv1.Cluster to allow someone to connect to
+// the kubernetes API. This test should fail if a new field is added to one of the structs without
+// updating the other.
+func TestCluster(t *testing.T) {
+	t.Parallel()
+
+	execType := reflect.TypeOf(Cluster{})
+	clientcmdType := reflect.TypeOf(clientcmdv1.Cluster{})
+
+	t.Run("exec cluster fields match clientcmd cluster fields", func(t *testing.T) {
+		t.Parallel()
+
+		// These are fields that are specific to Cluster and shouldn't be in clientcmdv1.Cluster.
+		execSkippedFieldNames := sets.NewString(
+			// Cluster uses Config to provide its cluster-specific configuration object.
+			"Config",
+		)
+
+		for i := 0; i < execType.NumField(); i++ {
+			execField := execType.Field(i)
+			if execSkippedFieldNames.Has(execField.Name) {
+				continue
+			}
+
+			t.Run(execField.Name, func(t *testing.T) {
+				t.Parallel()
+				clientcmdField, ok := clientcmdType.FieldByName(execField.Name)
+				if !ok {
+					t.Errorf("unknown field '%s'", execField.Name)
+				} else if execField.Type != clientcmdField.Type {
+					t.Error("type mismatch:", execField.Type, "!=", clientcmdField.Type)
+				} else if execField.Tag != clientcmdField.Tag {
+					t.Error("tag mismatch:", execField.Tag, "!=", clientcmdField.Tag)
+				}
+			})
+		}
+	})
+
+	t.Run("clientcmd cluster fields match exec cluster fields", func(t *testing.T) {
+		t.Parallel()
+
+		// These are the fields that we don't want to shadow from clientcmdv1.Cluster.
+		clientcmdSkippedFieldNames := sets.NewString(
+			// CA data will be passed via CertificateAuthorityData, so we don't need this field.
+			"CertificateAuthority",
+			// Cluster uses Config to provide its cluster-specific configuration object.
+			"Extensions",
+		)
+
+		for i := 0; i < clientcmdType.NumField(); i++ {
+			clientcmdField := clientcmdType.Field(i)
+			if clientcmdSkippedFieldNames.Has(clientcmdField.Name) {
+				continue
+			}
+
+			t.Run(clientcmdField.Name, func(t *testing.T) {
+				t.Parallel()
+				execField, ok := execType.FieldByName(clientcmdField.Name)
+				if !ok {
+					t.Errorf("unknown field '%s'", clientcmdField.Name)
+				} else if clientcmdField.Type != execField.Type {
+					t.Error("field mismatch:", clientcmdField.Type, "!=", execField.Type)
+				} else if clientcmdField.Tag != execField.Tag {
+					t.Error("tag mismatch:", clientcmdField.Tag, "!=", execField.Tag)
```

Comment:
be explicit about what the expected action is here... typically, it will be to fix the exec cluster config type

Reply:
77da453042f